lib/spack/spack/util/gpg.py

```diff
@@ -63,7 +63,7 @@ def init(gnupghome=None, force=False):
     # Set the executable objects for "gpg" and "gpgconf"
     with spack.bootstrap.ensure_bootstrap_configuration():
         spack.bootstrap.ensure_gpg_in_path_or_raise()
-        GPG, GPGCONF = _gpg(), _gpgconf()
+    GPG, GPGCONF = _gpg(), _gpgconf()
```

Comment:
(question) Were you seeing a problem with this when it was inside the context manager? Overall, after reviewing uses of `ensure_bootstrap_configuration`, they seem inconsistent to me, so I'm not actually sure which is strictly correct. Generally, the original organization might be more robust:

1. `ensure_bootstrap_configuration` disables any active spack envs,
2. For now, re/activating the `Environment` object does not modify `PATH` the same way that `spack env activate` does (so what you did should generally be OK)
3. `ensure_gpg_in_path_or_raise` will prepend to PATH if an installation occurs as part of meeting its guarantee

if [2] changes, this change could be problematic (i.e. would interfere with [3]). I also wouldn't have expected the original organization to cause you any trouble.

Reply:
Oh I actually thought this _was_ restoring it to previous behavior, I couldn't remember if I had initially moved it under the context manager during the original approach. Should have checked the diff, will restore.